crypto/evp/evp_enc.c

```diff
@@ -142,6 +142,7 @@ int EVP_CipherInit_ex(EVP_CIPHER_CTX *ctx, const EVP_CIPHER *cipher,
 
     if (tmpcipher->prov == NULL) {
         switch(tmpcipher->nid) {
+        case NID_undef:
```

Comment:
Why do we still have this switch?  Wouldn't it be more sensible (and less error prone, because *you* might remember to fill this in, but others might not) to try to fetch, and only fall back to legacy if that fails?

might be subject for another PR...

Reply:
Another PR I'd say

Reply:
Okie